src/wrapper/jnienv.rs

```diff
@@ -78,7 +78,7 @@ use crate::{
 #[derive(Clone, Copy)]
 #[repr(transparent)]
 pub struct JNIEnv<'a> {
-    internal: *mut sys::JNIEnv,
+    pub(crate) internal: *mut sys::JNIEnv,
```

Comment:
oh, how come this has been made public?

I would expect to leave this as private since it should be read-only - accessible via `get_raw()` if you want to do something `unsafe` with the raw pointer or `into_raw`

Reply:
Oh damn, missed the get_raw 😅 will change that back

Reply:
There seems to be no `get_raw` on JNIEnv. However, rather than making one, might it not be better to move the `get_string_utf_chars` back into JNIEnv, but change the visibility. This means keeping all JNI calls inside JNIEnv, then invoke them from JavaStr.

Reply:
Sorry I don't follow what the concern is here.

In this context isn't it enough that there's (already) a `get_raw()` for `JavaStr` in case an app has some `unsafe` thing it needs to do with the raw pointer that's stored in a `JavaStr`

There doesn't need to be any `get_raw` API under `JNIEnv` for strings does there?

Reply:
Currently (from what I've seen), all 'real' JNI calls are done in JNIEnv, this PR would move both string_utf_chars functions out of there into JavaStr (counting #372 as well). Would it not be better to leave them in `JNIEnv`, but make them non-pub, and 'expose' their functionality via JavaStr.

For `get_raw`, I couldn't find one on `JNIEnv` (to get at the inner pointer), hence one would need to be made if we want the 'real' JNI call to remain in `JavaStr`

Reply:
Ah, I see.

I guess either is reasonable though I personally like the encapsulation/locality from moving these functions into `JavaStr` if that's going to become the one and only thing that uses these.

Adding a `get_raw` for the env also sounds good , it's a bit surprised that's missing. Or else the env inner pointer could be made `pub(crate)` perhaps.

Reply:
I'll go with keeping it in JavaStr then, and add a get_raw to JNIEnv

Reply:
okey, cool, sounds good to me